Source/WTF/wtf/DataLog.cpp

```diff
@@ -34,16 +34,29 @@
 #include <wtf/ProcessID.h>
 #include <mutex>
 
+// Setting DATA_LOG_TO_FILE to 1 will cause logs to be sent to the filename
+// specified in the WTF_DATA_LOG_FILENAME envvar.
 #define DATA_LOG_TO_FILE 0
-
-// Set to 1 to use the temp directory from confstr instead of hardcoded directory.
-// The last component of DATA_LOG_FILENAME will still be used.
+// Alternatively, setting this to 1 will override the above settings and use
+// the temp directory from confstr instead of the hardcoded directory.
 #define DATA_LOG_TO_DARWIN_TEMP_DIR 0
 
-// Uncomment to force logging to the given file regardless of what the environment variable says.
+// Setting DATA_LOG_TO_FILE_IGNORE_ENVVAR to 1 will cause both data-log options
+// above to always use the fallback filename.
+#define DATA_LOG_IGNORE_ENV_VAR 0
+
+static_assert(!(DATA_LOG_TO_FILE && DATA_LOG_TO_DARWIN_TEMP_DIR), "Set at most one data-log file target");
```

Comment:
Shouldn't this be an ifdef, something like
```
#if defined(DATA_LOG_TO_FILE) && defined(DATA_LOG_TO_DARWIN_TEMP_DIR)
#error "Set at most one data-log file target"
#endif
```

Reply:
My bad, I'll put up a quick patch to fix it.

Reply:
https://github.com/WebKit/WebKit/pull/29469